### PR TITLE
JPERF-126: HTTP load

### PR DESCRIPTION
**Pull request by dagguh**

### CODE REVIEW

**Comment by dagguh** [comment]
- [] CHANGELOG

**Comment by dagguh** [comment]
@mgrzaslewicz note 8cf468c, we should do the same with REST Actions internally

**Comment by dagguh** [comment]
FIxed all the tests, @mgrzaslewicz it's ready for review.

File: src/main/kotlin/com/atlassian/performance/tools/virtualusers/LoadTest.kt [hunk 1]
```diff
@@ -1,251 +1,84 @@
...
-                WebDriverDiagnostics(this, this, diagnoses),
-                diagnosisPatience
-            ),
-            diagnosisLimit
-        )
+        logger.info("Thread containers closed")
     }
 }
+
+private class ContainedThread(
```
Comment (mgrzaslewicz) [inline src/main/kotlin/com/atlassian/performance/tools/virtualusers/LoadTest.kt:81]: this name alone does not say much. How about `LoadThreadWithContainer`?

File: src/main/kotlin/com/atlassian/performance/tools/virtualusers/load/HttpLoadProcess.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    override fun run() {
+        meter.measure(REST_SEARCH) {
+            val request = Request.Builder()
+                .post(RequestBody.create(mediaTypeJson, requestJson.toString()))
+                .url(base.resolve("rest/api/2/search").toURL())
+                .build()
+            val response = http.newCall(request).execute()
+            if (response.isSuccessful.not()) {
+                throw Exception("Response failed: $response")
```
Comment (mgrzaslewicz) [inline src/main/kotlin/com/atlassian/performance/tools/virtualusers/load/HttpLoadProcess.kt:88]: That's not enough information in the exception as often body contains the reason why call failed.
Draft without closing the stream:
                throw Exception("Response failed: $response. Body: ${response.body()?.string()}")

**Review by mgrzaslewicz — APPROVED** [review]
@dagguh I've disabled auto merge so you could see the latest change plus one unresolved comment

**Review by mgrzaslewicz — APPROVED** [review]

**Pull request merged**